pkg/reconciler/v1alpha1/broker/broker.go

```diff
@@ -80,94 +88,99 @@ type ReconcilerArgs struct {
+	serviceInformer corev1informers.ServiceInformer,
+	deploymentInformer appsv1informers.DeploymentInformer,
+	args ReconcilerArgs,
+) *controller.Impl {
+
+	r := &Reconciler{
+		Base:                      reconciler.NewBase(opt, controllerAgentName),
+		brokerLister:              brokerInformer.Lister(),
+		channelLister:             channelInformer.Lister(),
+		serviceLister:             serviceInformer.Lister(),
+		deploymentLister:          deploymentInformer.Lister(),
+		subscriptionLister:        subscriptionInformer.Lister(),
+		ingressImage:              args.IngressImage,
+		ingressServiceAccountName: args.IngressServiceAccountName,
+		filterImage:               args.FilterImage,
+		filterServiceAccountName:  args.FilterServiceAccountName,
+	}
+	impl := controller.NewImpl(r, r.Logger, ReconcilerName, reconciler.MustNewStatsReporter(ReconcilerName, r.Logger))
+
+	r.Logger.Info("Setting up event handlers")
+
+	brokerInformer.Informer().AddEventHandler(reconciler.Handler(impl.Enqueue))
+
+	channelInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Broker")),
+		Handler:    reconciler.Handler(impl.EnqueueControllerOf),
+	})
 
-		// Watch all the resources that the Broker reconciles.
-		for _, t := range []runtime.Object{&v1alpha1.Channel{}, &corev1.Service{}, &v1.Deployment{}} {
-			err = c.Watch(&source.Kind{Type: t}, &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.Broker{}, IsController: true})
-			if err != nil {
-				return nil, err
-			}
-		}
+	serviceInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Broker")),
+		Handler:    reconciler.Handler(impl.EnqueueControllerOf),
+	})
 
-		return c, nil
-	}
-}
+	deploymentInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Broker")),
+		Handler:    reconciler.Handler(impl.EnqueueControllerOf),
+	})
 
-// InjectClient implements controller runtime's inject.Client.
-func (r *reconciler) InjectClient(c client.Client) error {
-	r.client = c
-	return nil
+	return impl
 }
 
 // Reconcile compares the actual state with the desired, and attempts to
 // converge the two. It then updates the Status block of the Broker resource
 // with the current status of the resource.
-func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	ctx := context.TODO()
-	ctx = logging.WithLogger(ctx, r.logger.With(zap.Any("request", request)))
-
-	broker := &v1alpha1.Broker{}
-	err := r.client.Get(ctx, request.NamespacedName, broker)
-
-	if k8serrors.IsNotFound(err) {
-		logging.FromContext(ctx).Info("Could not find Broker")
-		return reconcile.Result{}, nil
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
```

Comment:
Save the key into the context. 

```golang
ctx = logging.WithLogger(ctx, r.Logger.Desugar().With(zap.String("key", key)))
```

Then remove it from all the other log statements.

Reply:
Done